src/color.rs

```diff
@@ -316,19 +316,19 @@ fn detect_color(conin: &mut dyn ConsoleRead, color: u16) -> io::Result<bool> {
     let mut total_read = 0;
 
     let timeout = Duration::from_millis(CONTROL_SEQUENCE_TIMEOUT);
-    let now = SystemTime::now();
     let bell = 7u8;
 
     // Either consume all data up to bell or wait for a timeout.
-    while buf[0] != bell && now.elapsed().unwrap() < timeout {
-        match conin.read(&mut buf) {
-            Ok(b) => total_read += b,
-            // Don't error out on a would block- this just means no response since async.
-            Err(err) if err.kind() == io::ErrorKind::WouldBlock => {}
-            Err(err) => return Err(err),
+    if conin.poll_timeout(timeout) {
+        while buf[0] != bell {
```

Comment:
I think this while loop being inside the poll is a potential issue (one that probably won't come up often but could).  Since if the poll succeeds then the while will call read.  I guess read is non blocking so would return the WouldBlock error (see comment below) but this could just spin until something showed up on the tty and then it would disappear in the void and this would fail (unless it happened to be a bel but at this point that is not likely).  I think you may need to incorporate the poll into the while condition (?) that way then while loop will not spin even in a degenerate case.  This might not handle the timeout 100% if something else came in via the tty other then what we want though...  Technically the timeout should be decremented by the time already waited OR maybe we should just remove the while loop and if the next thing in was not the bel then fail...  If other data is coming in on the tty then this could be an issue BUT I think this will consume and discard that data anyway so basically if you try to detect a color while the tty is writing stuff you are kinda hosed either way so maybe we just hose things in the simpler direction... 

Another wrinkle here, if conin has data queued before the escape code for this is written then there will be data already in the way (probably the most likely way to trip this up).  I think the only way to detect this currently would be to add a poll_timeout with a really small timeout before the write and return an error if data is ready to read (someone is being real sloppy...).  I kinda hate this though because this call will always be slowed down by the delay with potentially a mostly unneeded syscall to boot.  It may just be better to go with erroring out if it reads something other then the bel in the timeout (this probably means that someone called this with data queued in conin which is not good).  I think I like this better and hopefully a dev would see that error during dev and make their code not so shitty...

In short I think we should get rid of the while and just return an error if it reads something other then bel in the timeout but open to discussion.